contracts/Staking.sol

```diff
@@ -195,6 +196,7 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable {
         uint256 eraNumber = IEraManager(settings.getContractAddress(SQContracts.EraManager)).safeUpdateAndGetEra();
         _reflectStakingAmount(eraNumber, delegation[_source][_indexer]);
         _reflectStakingAmount(eraNumber, totalStakingAmount[_indexer]);
+        IStakingAllocation(settings.getContractAddress(SQContracts.StakingAllocation)).update(_indexer, totalStakingAmount[_indexer].valueAt);
```

Comment:
we only swap EraValue struct here, it doesn't mean any value changes. I think it shouldn't call stakingallocation.update